src/prefect/tasks/azure/cosmosdb.py

```diff
@@ -0,0 +1,313 @@
+        - document_or_container_link (str, optional): link to a document or container. 
+            If a document link is provided, the document in question is returned, otherwise 
+            all docuements are returned.
+        - azure_credentials_secret (str, optional): the name of the Prefect Secret
+            that stores your Azure credentials; this Secret must be JSON string with the key
+            `AZ_COSMOS_AUTH`. The value should be dictionary containing `masterKey` or `resourceTokens`, 
+            where the `masterKey` value is the default authorization key to use to
+            create the client, and `resourceTokens` value is the alternative
+            authorization key.
+        - options (dict, optional): options to be passed to the 
+            `azure.cosmos.cosmos_client.CosmosClient.ReadItem` or `ReadItems` method.
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        url: str = None,
+        document_or_container_link: str = None,
+        azure_credentials_secret: str = "AZ_CREDENTIALS",
+        options: Dict[Any, Any] = None,
+        **kwargs
+    ) -> None:
+        self.url = url
+        self.document_or_container_link = document_or_container_link
+        self.azure_credentials_secret = azure_credentials_secret
+        self.options = options
+
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs(
+        "url", "document_or_container_link", "azure_credentials_secret", "options"
+    )
+    def run(
+        self,
+        url: str = None,
+        document_or_container_link: str = None,
+        azure_credentials_secret: str = "AZ_CREDENTIALS",
+        options: Dict[Any, Any] = None,
+    ) -> Union[Dict[Any, Any], QueryIterable]:
+        """
+        Task run method.
+
+        Args:
+            - url (str, optional): The url to the database.
+            - document_or_container_link (str, optional): link to a document or container. 
+                If a document link is provided, the document in question is returned, otherwise 
+                all docuements are returned.
+            - azure_credentials_secret (str, optional): the name of the Prefect Secret
+                that stores your Azure credentials; this Secret must be JSON string with the key
+                `AZ_COSMOS_AUTH`. The value should be dictionary containing `masterKey` or `resourceTokens`, 
+                where the `masterKey` value is the default authorization key to use to
+                create the client, and `resourceTokens` value is the alternative
+                authorization key.
+            - options (dict, optional): options to be passed to the 
+                `azure.cosmos.cosmos_client.CosmosClient.ReadItem` or `ReadItems` method.
+
+        Returns:
+            - (dict or azure.cosmos.query_iterable.QueryIterable)): a single document or all documents. 
```

Comment:
Is the `QueryIterable` object an iterator or a fully hydrated list?  Ultimately I'm wondering if you happen to know whether it's `cloudpickle`able.

Reply:
Good catch! I changed `CosmosDBReadItems` and `CosmosDBQueryItems` to return lists, following the example in `prefect.tasks.google.bigquery.BigQueryTask`